conda/core/index.py

```diff
@@ -100,6 +102,54 @@ def supplement_index_with_cache(index, channels):
         index[dist] = rec
 
 
+def supplement_index_with_repodata(index, repodata, channel, priority):
+    repodata_info = repodata.get('info', {})
+    arch = repodata_info.get('arch')
+    platform = repodata_info.get('platform')
+    schannel = channel.canonical_name
+    channel_url = channel.url()
+    auth = channel.auth
+    for fn, info in iteritems(repodata['packages']):
+        rec = IndexRecord.from_objects(info,
+                                       fn=fn,
+                                       arch=arch,
+                                       platform=platform,
+                                       schannel=schannel,
+                                       channel=channel_url,
+                                       priority=priority,
+                                       url=join_url(channel_url, fn),
+                                       auth=auth)
+        dist = Dist(rec)
+        index[dist] = rec
+        if 'with_features_depends' in info:
+            base_deps = info.get('depends', ())
+            base_feats = set(info.get('features', '').strip().split())
+            for feat, deps in iteritems(info['with_features_depends']):
+                feat = set(feat.strip().split())
+                snames = {MatchSpec(s).name for s in deps}
+                base2 = [s for s in base_deps if MatchSpec(s).name not in snames]
+                feat2 = ' '.join(sorted(base_feats | feat))
+                feat = ' '.join(sorted(feat))
+                deps2 = base2 + deps
+                dist = Dist.from_objects(dist, with_features_depends=feat)
+                rec2 = IndexRecord.from_objects(rec, features=feat2, depends=deps2)
+                index[dist] = rec2
+
+
+def supplement_index_with_features(index, features=()):
```

Comment:
This function is needed so that some sort of index entry exists for each `context.features` entry. Otherwise, the `Resolve` object has to create these metapackages itself.